### ISSUE TITLE
TalkBack and VoiceOver issues with sharing code

### ISSUE THREAD

**Issue by denisedresbach** [covid-alert-ny/covid-green-app#618]
App version: 1.1.1.64
OS: iOS 14.2, Android 11
Device: iPhone SE, Pixel 3 XL
Reproducible: Always

Summary: Here is a list of issues I ran into with VoiceOver and TalkBack when looking at https://github.com/covid-alert-ny/covid-green-app/issues/612

Screen- Share Your Close Contact Codes with Screen Reader enabled

1. Android/iOS - when first accessing this screen, the screen reader goes straight to the Edit box and doesn't read any of the text on the screen. I would expect it to start at the very beginning.
2. Android - Enter max amount of characters allowed (invalid code). If you try to enter more, the screen reader just repeats the numbers you are entering instead of saying that you reached the limit.
3. Android - Enter invalid code that is 8 characters long and try to go to button to enter. Results: Cannot access that button with screen reader or manually tapping it.
4. Android - Enter a valid code. Results: Keyboard disappears and I hear it say that max length was entered. I cannot tap on the box to edit. I guess I don't need to if it is valid, but could possibly confuse the user since all it says is max length entered. Workaround is to have it read next paragraph. 

No. 3 screenshot ![Screenshot_20210118-160619](https://user-images.githubusercontent.com/69995161/104981573-b8db4180-59bd-11eb-807b-72cd71411a1f.png)

**Comment by AlanSl** [comment]
Should all be fixed in the next build. We can't fine control how TalkBack responds to reaching the end of input but the behaviour will be different now and overall should be smoother.

**Comment by denisedresbach** [comment]
Smoother now 1.1.1.67 but I have comments below. @AlanSl 

Original list of issues from above
1. PASSED iOS and Android - now reads heading first and not text field
2. This hasn't changed but hopefully with a valid code it will just move to next item? I don't have access to valid codes that I can type in right now so I cannot fully test.
3. PASSED iOS and Android - I can now skip to next button and tap it to enter a code (button is called OK)
I cannot test this completely as I no longer have access to valid 8 char codes.
4. I cannot test this because I don't have valid fields but it sounds like from your comment we can't control it?

Couple of issues I noticed and can write up new ticket or just keep this open:

iOS
1. If I submit and invalid code it would be nice if the error is read to me (Android does). What it does now is says "code input text field is editing" then it reads the code to me along with some other instructions about correcting misspelled words.
2. https://github.com/covid-alert-ny/covid-green-app/issues/629 Makes it a little awkward when in screen reading mode 
3. Cannot fully test this feature because of access to valid codes

**Comment by AlanSl** [comment]
@denisedresbach I can't replicate the issue with the error not being read on iOS, for me it's the same as Android, after a moment it focuses and reads the error, see video here:

https://share.icloud.com/photos/0dDnYitNDXngI2KBZP4BZpoxg

What does it do for you exactly, and are you doing anything different on entering the code?

**Comment by denisedresbach** [comment]
@AlanSl Try entering 6 codes and then ok. Does it read the error then?

**Comment by AlanSl** [comment]
@denisedresbach Yes, and 8 digits is the same

**Comment by denisedresbach** [comment]
https://user-images.githubusercontent.com/69995161/105379513-c19c6500-5bc1-11eb-9456-c32e62d7b335.mp4

@AlanSl I type 6 or 8 char and then **swipe right** to go to the ok button. Then I double tap to say ok.

**Comment by AlanSl** [comment]
It's possible that the change in keyboard type did it, let's test this again in the next build.

**Comment by denisedresbach** [comment]
1.1.1.69 tested with a couple issues still coming up for me. I am going to close this and write them up separately 

- [x]  - Verify reads heading first and not text field
- [x] Enter max amount of characters - 
        Android -when entering a valid code it says "keyboard hidden" Better for it to tell user it was accepted, workaround is to read the screen back to get new info and move to next 
**---write up new ticket for future (iOS read next screen automatically)**
        iOS -
- [x]  - Verify I can skip to next button and tap it to enter a code (button is called OK)

- [x]  Android Typed invalid code and I can tap done on keyboard and then I hear her read the ok button. double tap to activate and invalid code message is shown.

iOS

- [Fail] If I submit and invalid code it would be nice if the error is read to me (Android does). What it does now is says "code input text field is editing" then it reads the code to me along with some other instructions about correcting misspelled words. 1.1.1.69 still happens for me. **I will write up a new ticket for it for future.**

- [x] #629 Makes it a little awkward when in screen reading mode --  628 was fixed

- [x] Cannot fully test this feature because of access to valid codes - I have valid codes now
